handlers.go

```diff
@@ -114,13 +87,26 @@ func getStateHandler(e *Enclave) http.HandlerFunc {
 // trusted enclave application.
 func putStateHandler(e *Enclave) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		body, err := io.ReadAll(newLimitReader(r.Body, maxKeyMaterialLen))
+		keys, err := io.ReadAll(newLimitReader(r.Body, maxKeyMaterialLen))
 		if err != nil {
 			http.Error(w, errFailedReqBody.Error(), http.StatusInternalServerError)
 			return
 		}
-		e.SetKeyMaterial(body)
+		e.keys.setAppKeys(keys)
 		w.WriteHeader(http.StatusOK)
+
+		// The leader's application keys have changed.  Re-synchronize the key
+		// material with all registered workers.  If synchronization fails for a
+		// given worker, unregister it.
+		elog.Printf("Application keys have changed.  Re-synchronizing with %d worker(s).",
+			e.workers.length())
+		go e.workers.forAll(
+			func(worker *url.URL) {
+				if err := asLeader(e.keys.get()).syncWith(worker); err != nil {
```

Comment:
verified this kicks off the full key sync protocol again and checks attestations as expected.